src/libs/rest_api/webview_server.cpp

```diff
@@ -0,0 +1,264 @@
+
+/*  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Library General Public License for more details.
+ *
+ *  Read the full text in the LICENSE.GPL file in the doc directory.
+ */
+
+#include "webview_server.h"
+
+#include "clips_rest_api/clips-rest-api.h"
+#include "rest_processor.h"
+#include "service_browse_handler.h"
+
+#include <config/config.h>
+#include <core/exceptions/system.h>
+#include <core/version.h>
+#include <logging/logger.h>
+#include <netcomm/service_discovery/service_browser.h>
+#include <netcomm/service_discovery/service_publisher.h>
+#include <sys/wait.h>
+#include <utils/misc/string_conversions.h>
+#include <utils/system/file.h>
+#include <utils/system/hostinfo.h>
+#include <webview/nav_manager.h>
+#include <webview/page_reply.h>
+#include <webview/request_dispatcher.h>
+#include <webview/request_manager.h>
+#include <webview/rest_api_manager.h>
+#include <webview/server.h>
+#include <webview/url_manager.h>
+
+#include <clipsmm.h>
+
+using namespace fawkes;
+
+/** @class WebviewServer "webview_server.h"
+ * Webview Server.
+ * This thread runs the HTTP server and handles requests via the
+ * WebRequestDispatcher.
+ * @author Tim Niemueller
+ */
+
+/** Constructor.
+ * @param enable_tp true to enable thread pool setting the thread to
+ * wait-for-wakeup mode, falso to run request processing in this
+ * thread.
+ */
+namespace llsfrb {
+
+WebviewServer::WebviewServer(bool                         enable_tp,
+                             fawkes::NetworkNameResolver *nnresolver,
+                             fawkes::ServicePublisher *   service_publisher,
+                             fawkes::ServiceBrowser *     service_browser,
+                             fawkes::Mutex &              clips_mutex,
+                             CLIPS::Environment *         env,
+                             Configuration *              config,
+                             Logger *                     logger)
+: fawkes::Thread("WebviewServer", Thread::OPMODE_CONTINUOUS)
+{
+	cfg_use_thread_pool_ = enable_tp;
+
+	if (!enable_tp)
+		set_prepfin_conc_loop(true);
+
+	nnresolver_        = nnresolver;
+	service_publisher_ = service_publisher;
+	service_browser_   = service_browser;
+	config_            = config;
+	logger_            = logger;
+
+	url_manager_      = new WebUrlManager();
+	request_manager_  = new WebRequestManager();
+	rest_api_manager_ = new WebviewRestApiManager();
+	clips_rest_api_   = new ClipsRestApi(env, clips_mutex, rest_api_manager_, logger_);
```

Comment:
Also here, I don't think any of those need to be raw pointers.

Reply:
Done, Also removed all pointers in libs\rest-api